### ISSUE TITLE
NEOPIXEL Setup

### ISSUE THREAD

**Issue by FabioMC** [makerbase-mks/MKS-Robin-Nano-V3.X#7]
I have tried different PINS not in use (PA13 and PC4) to setup the Neopixel led's of my printer but no success so far. What should be the best PIN option to setup those?

**Comment by Aritzobixo** [comment]
It would be interesting to know how to connect a led strip. It is a pity that this good board does not have a neopixel connection but it is possible to connect it to other great pins.

**Comment by Aritzobixo** [comment]
> He probado diferentes PIN que no están en uso (PA13 y PC4) para configurar los LED de Neopixel de mi impresora, pero hasta ahora no he tenido éxito. ¿Cuál debería ser la mejor opción de PIN para configurarlos?

have you modified the .pins file?

**Comment by FabioMC** [comment]
Yes. Have modified but no success. The output pins are limited. I found that one option is to use the Power_Off pin but than I will loose the configuration to turn off the printer when the print is finish. I would like an alternative for it.

**Comment by Aritzobixo** [comment]
It would be nice if you tell us something about it. Because otherwise we lose

**Comment by Aritzobixo** [comment]
@MKS-Sean any solution to the problem?

**Comment by MKS-Sean** [comment]
@FabioMC @Aritzobixo Hello, PA8 (BL_TOUCH signal pin) or TX and RX pins of UART3 can achieve this function.

**Comment by Aritzobixo** [comment]
![IMG_20210224_115940](https://user-images.githubusercontent.com/77989897/108992001-0e7bcb80-7699-11eb-91c9-cc2e254bf274.jpg)
and what pin is S, G, 5v ???

**Comment by MKS-Sean** [comment]
S is PA8 (signal pin), you can see the pin number on the back of the motherboard

**Comment by Aritzobixo** [comment]
> @FabioMC @Aritzobixo Hola, PA8 (pin de señal BL_TOUCH) o los pines TX y RX de UART3 pueden lograr esta función.

but that's for bltouch and i use bltouch too

**Comment by MKS-Sean** [comment]
In this case you can use uart3's tx or rx pin

**Comment by MKS-Sean** [comment]
![Uploading image.png…]()

**Comment by Aritzobixo** [comment]
but that's for bltouch and i use bltouch too

**Comment by Aritzobixo** [comment]
The image can not be seen

**Comment by Aritzobixo** [comment]
If I use E1 driver, the connections are S = PA3, G = GND, and 5v? @MKS-Sean @makerbase-mks

**Comment by FabioMC** [comment]
I would like to try TX RX UART 3 since looks more simple. What are the pin codes for those? PB10 and PB11? Is that correct?

**Comment by FabioMC** [comment]
Ok. Tried UART 3. I used PB10 since its near the GND so an easier way to fit the connector. Everything is working properly. Thanks for your kind feedback.

**Comment by Aritzobixo** [comment]
> Está bien. Probé UART 3. Usé PB10 ya que está cerca del GND, por lo que es una forma más fácil de colocar el conector. Todo funciona correctamente. Gracias por tus amables comentarios.

could you put a photo of the connections?  Thanks a lot

**Comment by FabioMC** [comment]
![IMG_3256](https://user-images.githubusercontent.com/29107577/109000339-580cec00-7682-11eb-98e3-a72577391b72.jpg)
Here it goes

**Comment by Aritzobixo** [comment]
> ![IMG_3256](https://user-images.githubusercontent.com/29107577/109000339-580cec00-7682-11eb-98e3-a72577391b72.jpg)
> Aquí va

there you lose touch screen, right?

**Comment by FabioMC** [comment]
Not really. I am using MKS-TS35 which uses only EXP1 and EXP2. I may not be able to use the new MKS H43 though. So far everything is working fine.

**Comment by Aritzobixo** [comment]
> Realmente no. Estoy usando MKS-TS35 que usa solo EXP1 y EXP2. Sin embargo, es posible que no pueda usar el nuevo MKS H43. Hasta ahora todo está funcionando bien.

Could you put the piece of code that you have added, please?

**Comment by FabioMC** [comment]
It is a very simple configuration. Enabled NEOPIXEL_LED and put the PIN number in front of NEOPIXEL_PIN. You don't need to change any other file

// Support for Adafruit NeoPixel LED driver
#define NEOPIXEL_LED
#if ENABLED(NEOPIXEL_LED)
  #define NEOPIXEL_TYPE   NEO_GRB // NEO_GRBW / NEO_GRB - four/three channel driver type (defined in Adafruit_NeoPixel.h)
  **#define NEOPIXEL_PIN    PB10    // LED driving pin**
  //#define NEOPIXEL2_TYPE NEOPIXEL_TYPE
  //#define NEOPIXEL2_PIN    5
  #define NEOPIXEL_PIXELS 32       // Number of LEDs in the strip. (Longest strip when NEOPIXEL2_SEPARATE is disabled.)
  #define NEOPIXEL_IS_SEQUENTIAL   // Sequential display for temperature change - LED by LED. Disable to change all LEDs at once.
  #define NEOPIXEL_BRIGHTNESS 127  // Initial brightness (0-255)
  #define NEOPIXEL_STARTUP_TEST  // Cycle through colors at startup

  // Support for second Adafruit NeoPixel LED driver controlled with M150 S1 ...
  //#define NEOPIXEL2_SEPARATE
  #if ENABLED(NEOPIXEL2_SEPARATE)
    #define NEOPIXEL2_PIXELS      15  // Number of LEDs in the second strip
    #define NEOPIXEL2_BRIGHTNESS 127  // Initial brightness (0-255)
    #define NEOPIXEL2_STARTUP_TEST    // Cycle through colors at startup
  #else
    //#define NEOPIXEL2_INSERIES      // Default behavior is NeoPixel 2 in parallel
  #endif

**Comment by Aritzobixo** [comment]
@FabioMC Hi Fabio, would you know how to do it with the uart pins? I do not understand where it says to connect the cables

**Comment by FabioMC** [comment]
His suggestion is to use the PIN PD8 into the E1 step motor jumpers. Please note that 5V and GND need to be connect to other place like the MT-DET2 or any other connector that you are not using.

![PD8 Pin](https://user-images.githubusercontent.com/29107577/109082300-74d90c00-76e2-11eb-8631-48f655ae83a8.jpg)

**Comment by mitofun** [comment]
thank you folks everything works! but MKS should do the instruction for this!))))  PB10 it is TX3 pin under the processor!

**Comment by rick090980** [comment]
Are you able to control the leds from the LCD with them set up this way, or only through g-code?

**Comment by Freddyb888** [comment]
How many leds is maximum you can connect?

**Comment by mitofun** [comment]
> How many leds is maximum you can connect?

hi) directly i think no more 8- if you ude the dcdc step down to 5 volts - when it is up to you)

**Comment by heatchclif** [comment]
what version of marlin did you use????

**Comment by Freddyb888** [comment]
> > How many leds is maximum you can connect?
> 
> hi) directly i think no more 8- if you ude the dcdc step down to 5 volts - when it is up to you)

I bought the DCdc step down how do I exa tly connect this and the 60 led ring? I dont want to break something. Thx

**Comment by kabodos** [comment]
which ui are u using, marlin  or TFT_LVGL_UI?

**Comment by Freddyb888** [comment]
> which ui are u using, marlin or TFT_LVGL_UI?

Tft
